window-purpose-x.el

```diff
@@ -653,13 +653,22 @@ This function removes the buffer denoted by BUFFER-OR-NAME from all
 window-local buffer lists."
   (interactive "bBuffer to replace: ")
   (let* ((buffer (window-normalize-buffer buffer-or-name))
-         (purpose (purpose-buffer-purpose buffer))
-         (other-buffers (delete buffer (purpose-buffers-with-purpose purpose))))
+         ;; Delay calculating other-buffers until we need it
+         ;; This prevents unnecessary calculations on temporary
+         ;; buffers created by `with-temp-buffer' and other likewise
+         ;; non-displayed buffers
+         (have-other-buffers nil)
+         (other-buffers nil))
     (dolist (window (window-list-1 nil nil t))
       (if (eq (window-buffer window) buffer)
           (unless (window--delete window t t)
-            (let ((dedicated (purpose-window-purpose-dedicated-p window))
-                  (deletable (window-deletable-p window)))
+            (let* ((purpose (purpose-buffer-purpose buffer))
+                   (other-buffers (if have-other-buffers
+                                      other-buffers
+                                    (setq have-other-buffers t
+                                          other-buffers (delete buffer (purpose-buffers-with-purpose purpose)))))
+                   (dedicated (purpose-window-purpose-dedicated-p window))
+                   (deletable (window-deletable-p window)))
```

Comment:
Instead of setting `other-buffers` and `have-other-buffers` inside the `let*` block, please set them with `setq` outside the `let*` block, i.e:
```elisp
(unless (window--delete window t t)
  (setq other-buffers ...
        have-other-buffers t)
  (let ((dedicated (purpose-window-purpose-dedicated-p window))
        (deletable (window-deletable-p window)))
    ...
```

Reply:
Ah yeah I don't know why I wrote it so confusing like that. Sorry.
I need it inside the `let*` since I need the value of `purpose`, but have written a much simpler version:
```lisp
(unless other-buffers-calculated
  (setq other-buffers (delete buffer (purpose-buffers-with-purpose purpose))
        other-buffers-calculated t))
```
Thank you. Let me know if that's good